### ISSUE TITLE
Virtual Network Service Endpoints for Azure CosmosDB

### ISSUE THREAD

**Issue by idgazit** [terraform-providers/terraform-provider-azurerm#1342]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add the ability to set Virtual Network Service Endpoints for Azure CosmosDB, in addition to the ip_range_filter

also to be able to configure it as a resource (external to the cosmos_db_account)

https://azure.microsoft.com/en-us/blog/virtual-network-service-endpoints-for-azure-cosmos-db-is-now-generally-available/

### New or Affected Resource(s)

* azurerm_cosmos_db_account

### Potential Terraform Configuration

```hcl
resource "azurerm_cosmosdb_account" "db" {
     ....
      vitual_networks = ""
}

resource "azurerm_cosmosdb_firewall_rule" "db_fw" {

}
```

**Comment by CryptonZylog** [comment]
Definitely would be useful!

For anyone else looking for the ARM template "crutches" at the moment, ARM has this available via `properties`:

```json
    "isVirtualNetworkFilterEnabled": true, 
    "virtualNetworkRules": [
      {
        "id": "/subscriptions/guid/resourceGroups/name/providers/Microsoft.Network/virtualNetworks/vnetname/subnets/subnetname"
      }
    ],
```

**Comment by janlunddk** [comment]
Really looking forward to this feature.
Minor detail : 
The resource should probably be called azurerm_cosmosdb_virtual_network_rule, just like the vnet service end point for sql server is called azurerm_sql_virtual_network_rule

**Comment by tombuildsstuff** [comment]
hey @idgazit @CryptonZylog @janlunddk 

Just to let you know that support for this has now landed in v1.16 of the AzureRM Provider (here's the full changelog](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.16.0/CHANGELOG.md)). You can upgrade to to this version by specifying it in your provider block:

```
provider "azurerm" {
  version = "=1.16.0"
}
```

... and then running `terraform init -upgrade` :)

Thanks!
